packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -970,3 +1061,12 @@ function extendBuildSpec(buildSpec: any, extend: any) {
     phase.commands.push(...extend.phases[phaseName].commands);
   }
 }
+
+function resourcePolicyStatementForRepository() {
```

Comment:
I would name this `ecrAccessForCodeBuildService` or something like that.

Also, can you add the return type to this function?

Reply:
Done